Sources/NestKit/GitHub/GitHubAssetRegistryClient.swift

```diff
@@ -46,6 +46,43 @@ public struct GitHubAssetRegistryClient: AssetRegistryClient {
         }
         return AssetInformation(tagName: assetResponse.tagName, assets: assets)
     }
+
+    public func fetchAssetsApplyingExcludedVersions(repositoryURL: URL, version: GitVersion, excludingVersions: [String]) async throws -> AssetInformation {
+        let assetURL = try GitHubURLBuilder.assetURL(repositoryURL, version: version, hasExcludedVersion: true)
+        var request = HTTPRequest(url: assetURL)
+        request.headerFields = [
+            .accept: "application/vnd.github+json",
+            .gitHubAPIVersion: "2022-11-28",
+        ]
+        guard let repositoryHost = repositoryURL.host() else {
+            fatalError("Unknown host")
+        }
+        if let config = registryConfigs?.config(for: repositoryURL) {
+            logger.debug("GitHub token for \(repositoryHost) is passed from \(config.environmentVariable)")
+            request.headerFields[.authorization] = "Bearer \(config.token)"
+        } else {
+            logger.debug("GitHub token for \(repositoryHost) is not provided.")
+        }
+
+        logger.debug("Request: \(repositoryURL)")
+        logger.debug("Request: \(request.headerFields)")
+        let (data, response) = try await httpClient.data(for: request)
+        logger.debug("Response: \(data.humanReadableJSONString() ?? "No data")")
+        logger.debug("Status: \(response.status)")
+
+        if response.status == .notFound {
+            throw AssetRegistryClientError.notFound
+        }
+        let assetResponses = try JSONDecoder().decode([GitHubAssetResponse].self, from: data)
+
+        guard let matchedAssetResponse = assetResponses.first(where: { !excludingVersions.contains($0.tagName) }) else {
+            throw AssetRegistryClientError.noMatchApplyingExcludedVersion
+        }
+
+        let assets = matchedAssetResponse.assets
+            .map { Asset(fileName: $0.name, url: $0.browserDownloadURL) }
+        return AssetInformation(tagName: matchedAssetResponse.tagName, assets: assets)
```

Comment:
`GitHubAssetRegistryClient` should focus on the network communication and encode/decode information.
Could you move this code that selects correct version to `NestFileController`?

Reply:
Thanks!

> Could you move this code that selects correct version to NestFileController?

It's possible.
Since the `fetchAssets` method also returns AssetInformation, how about creating another class to handle type conversion?
For example, something like `GitHubAssetLoader` or `GitHubAssetService`?